### ISSUE TITLE
[Featured Product] Secondary image alt creates verbose experience

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#764]
## Issue summary

When a featured product section has the "Show second image on hover" setting enabled, the secondary image features `alt` text. This could create a verbose user experience for screen reader users who do not have access to the secondary image.

**This also exists for any product cards and the multimedia collage section.**

<details>
  <summary>Screenshots</summary>

</details>

#### HTML

```html
<div class="media media--transparent media--square media--hover-effect">
  <img
    srcset="…"
    src="//cdn.shopify.com/s/files/1/0014/4972/1914/products/5-panel-hat_4ee20a27-8d5a-490e-a2fc-1f9c3beb7bf5_533x.jpg?v=1549913928"
    sizes="…"
    alt="5 Panel Camp Cap"
    loading="lazy"
    class="motion-reduce"
    width="480"
    height="480"
  />
  <img
    srcset="…"
    src="//cdn.shopify.com/s/files/1/0014/4972/1914/products/0010untitledAug2014_533x.jpg?v=1549913928"
    sizes="…"
    alt="5 Panel Camp Cap"
    loading="lazy"
    class="motion-reduce"
    width="1000"
    height="1500"
  />
</div>
```

## Steps to reproduce

1. Go to a Dawn theme store with any supported browser
2. Find a Featured product section with the "Show second image on hover" setting enabled
3. With an active screen reader, navigate through the content via virtual cursor

### Expected

- Only the visible image `alt` to be described aloud.

### Actual

- Both visible and hidden image `alt` are available for screen reader consumption.

## Recommendation

Remove the `alt` attribute value from the secondary image, setting this image as decorative.

Rationale for keeping this a **hover only** feature, _not_ showing on focus:

1. Content is unpredictable. Assuming the merchant sets the primary image as the "best" representation of the product, we'd want low-vision keyboard-only or zoom users to see this primary image on focus. If the image swapped to the secondary image on focus, the low-vision zoom user would potentially _only_ see the secondary image and not the primary image, resulting in this user missing out on reviewing the "best" representation of the product.
2. Cards are meant as a "quick look" – a way for the user to quickly inspect and gain a limited view on what the product _is_ before clicking into its landing page. If we keep each primary and secondary image `alt` text available, this causes an overly verbose user experience for screen reader users, slowing them down on navigating though this "quick look" content.

With this said, the ideal user experience for product cards with "Show second image on hover" enabled would be to _not_ display the second image on focus **and** remove the second image `alt` text, treating it as decorative. Thus, providing the best "quick look" content for low-vision zoom users and screen reader users (no change for sighted mouse users nor disruption of the intended design.)

#### HTML

```html
<div class="media media--transparent media--square media--hover-effect">
  <img
    srcset="…"
    src="//cdn.shopify.com/s/files/1/0014/4972/1914/products/5-panel-hat_4ee20a27-8d5a-490e-a2fc-1f9c3beb7bf5_533x.jpg?v=1549913928"
    sizes="…"
    alt="5 Panel Camp Cap"
    loading="lazy"
    class="motion-reduce"
    width="480"
    height="480"
  />
  <img
    srcset="…"
    src="//cdn.shopify.com/s/files/1/0014/4972/1914/products/0010untitledAug2014_533x.jpg?v=1549913928"
    sizes="…"
    alt=""
    loading="lazy"
    class="motion-reduce"
    width="1000"
    height="1500"
  />
</div>
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **Browser:** Any
- **Dawn version:** 2.2.0
- **Component:** Featured Product
- **WCAG Principal:** Perceivable
- **WCAG SC:** [1.1.1 Non-text Content](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html)
- **Severity:** Medium
- **Effort:** Low
